### PR TITLE
Add a flag to sky_shell that keeps the process alive after the Dart script completes

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @cbracken

File: shell/common/switches.h [hunk 1]
```diff
@@ -82,6 +82,9 @@ DEF_SWITCH(UseTestFonts,
            "will make font resolution default to the Ahem test font on all "
            "platforms (See https://www.w3.org/Style/CSS/Test/Fonts/Ahem/). "
            "This option is only available on the desktop test shells.")
+DEF_SWITCH(RunForever,
+           "run-forever",
+           "Keep the shell running after the Dart script has completed.")
```
Comment (abarth) [inline shell/common/switches.h]: We should say that this is the default for interactive but not the default for `--non-interactive` mode.

**Comment by abarth** [comment]
LGTM

**Comment by jason-simmons** [comment]
Updated the usage text and added the run-forever flag to the observatory test script

**Pull request merged**